docs/proposals/guidelines/proposal-template.md

```diff
@@ -0,0 +1,212 @@
+replaces:
+  - "/docs/proposals/that-less-than-great-idea.md"
+superseded-by:
+  - "/docs/proposals/our-past-effort.md"
+---
+
+# Neat Enhancement Idea
+
+This is the title of the enhancement. Keep it simple and descriptive. A good title can help
+communicate what the enhancement is and should be considered as part of any review.
+
+The YAML `title` should be lowercased and spaces/punctuation should be replaced with `-`.
+
+To get started with this template:
+
+1. **Make a copy of this template.** Copy this template into the main `proposals` directory.
+2. **Fill out the "overview" sections.** This includes the Summary and Motivation sections. These
+   should be easy and explain why the community should desire this enhancement.
+3. **Create a PR.** Assign it to folks with expertise in that domain to help sponsor the process.
+4. **Merge at each milestone.** Merge when the design is able to transition to a new status
+   (provisional, implementable, implemented, etc.). View anything marked as `provisional` as an idea
+   worth exploring in the future, but not accepted as ready to execute. Anything marked as
+   `implementable` should clearly communicate how an enhancement is coded up and delivered. Aim for
+   single topic PRs to keep discussions focused. If you disagree with what is already in a document,
+   open a new PR with suggested changes.
+
+The `Metadata` section above is intended to support the creation of tooling around the enhancement
+process.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [docs](/docs/)
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding to implement the
+design. For instance:
+
+> 1. This locks a build strategy to run privileged pods. Should we do this?
+
+## Summary
+
+The `Summary` section is incredibly important for producing high quality user-focused documentation
+such as release notes or a development roadmap. It should be possible to collect this information
+before implementation begins in order to avoid requiring implementors to split their attention
+between writing release notes and implementing the feature itself.
+
+A good summary is probably at least a paragraph in length.
+
+## Motivation
+
+This section is for explicitly listing the motivation, goals and non-goals of this proposal.
+Describe why the change is important and the benefits to users.
+
+### Goals
+
+List the specific goals of the proposal. How will we know that this has succeeded?
+
+### Non-Goals
+
+What is out of scope for this proposal? Listing non-goals helps to focus discussion and make
+progress.
+
+## Proposal
+
+This is where we get down to the nitty gritty of what the proposal actually is.
+
+### User Stories [optional]
```

Comment:
I would make this non-option, i.e. emphasize on how a user would face the proposal.

Reply:
Good point, the counter-perspective, would that make the barrier for entry as bit too high? Or, if someone is making a proposal, at this point, idea has been socialized well enough to seek more details?

Reply:
So, let's say 
STEP 1: A contributor creates an enhancement proposal.
STEP 2: Discussion with other contributors on the enhancement proposal.
STEP 3: Based on the feedback from other contributors, this enhancement proposal can be modified.
STEP 4: When there is an agreement that this proposal IS NOW GOOD TO BE IMPLEMENTED, after which the user story should be created. 

If the proposal is rejected, user stories should not be created.

Remember, this is a **PROPOSAL**
User stories are a result of this proposal when the proposal is accepted.

Reply:
I see it differently - a need to support a user story triggers the thinking about a proposal. What is a benefit of a proposal that cannot backup itself with user stories/new/improved functionality?